### ISSUE TITLE
@doc for function head is ignored when there is a defdelegate.

### ISSUE THREAD

**Issue by crabonature** [elixir-lang/ex_doc#819]
ex_doc: 18.1
elixir:  master

I got some problem I defined function head before all my function closures but ex_doc is ignoring `@doc` tag related to that head. It tries to get another closure documentation. Maybe `defdelegate` is a problem? Function head is taken properly for documentation and `@spec` are processed properly.

```elixir
defmodule Sample do
  @moduledoc """
  Some great documentation.
  """

  @doc """
  Documenting our great do_something function.
  This doc is ignored completly.
  """
  def do_something(thing1, thing2)

  # I can't defdelegate it because there is an atom
  @spec do_something(String.t(), :special_case) :: String.t()
  def do_something(thing1, :special_case), do: Sample.Impl.do_something(thing1, :special_case)

  @spec do_something(String.t(), :ordinary_case) :: String.t()
  def do_something(thing1, :ordinary_case), do: Sample.Impl.do_something(thing1, :ordinary_case)

  @doc """
  Always getting this one badly. 
  Even if there is no documentation here at all it redirects to Sample.Impl.do_something/2 in generated files.
  """
  @spec do_something(String.t(), String.t()) :: String.t()
  defdelegate do_something(thing1, just_another_thing), to: Sample.Impl
end
```

```elixir
defmodule Sample.Impl do
  @moduledoc false
  
  def do_something(thing1, :special_case), do:  "#{thing1} - so special"
  def do_something(thing1, :ordinary_case), do:  "#{thing1} - just ok"
  def do_something(thing1, thing2), do: "#{thing1} and #{thing2}"
end
```

![ex_doc_1](https://user-images.githubusercontent.com/1584721/34642442-56e8a92c-f313-11e7-8f78-87de560fdf74.png)

**Comment by josevalim** [comment]
Can you please show us the documentation that would show up in the browser if you remove the second `@doc` entry? And what do you get if you try to fetch the docs in IEx?

**Comment by crabonature** [comment]
So in IEx with both `@doc`:
![term_with](https://user-images.githubusercontent.com/1584721/34643091-928e1272-f31e-11e7-9e17-82ae5184ddd7.png)

in IEx without second `@doc`:
![term_without](https://user-images.githubusercontent.com/1584721/34643095-aa116b10-f31e-11e7-922f-c2e2cf28dae2.png)

generated doc without second `@doc`:
![ex_doc_2](https://user-images.githubusercontent.com/1584721/34643097-b8f67260-f31e-11e7-91e2-1649a41303e0.png)

**Comment by josevalim** [comment]
Fixed on Elixir master and v1.6 branches.

**Comment by crabonature** [comment]
Fast one. Thank you ❤️
